chain/chain/src/garbage_collection.rs

```diff
@@ -1031,3 +1082,129 @@ impl<'a> ChainStoreUpdate<'a> {
         self.merge(store_update);
     }
 }
+
+/// Returns shards that we tracked in a past epoch, given hash of a block in the epoch.
+/// The block has to be available, so this function has to be called before gc is run for the block.
+///
+/// Note that validator ID or shard tracking config could have change since the epoch,
+/// so we have to rely on what is stored in the database to figure out tracked shards.
+/// We rely on `TrieChanges` column to preserve what shards this node tracked at that time.
+fn get_tracked_shards_in_past_epoch(
+    chain_store_update: &ChainStoreUpdate,
+    epoch_manager: &Arc<dyn EpochManagerAdapter>,
+    past_epoch_block_hash: &CryptoHash,
+) -> Result<Vec<ShardUId>, Error> {
```

Comment:
I don't think this method relies on the fact that the block is in past epoch or that it is the last block of that epoch. I would suggest removing `past_epoch` from the method method and argument names.